scripts/run-with-tarball.sh

```diff
@@ -93,11 +80,33 @@ if [ -z "${LLVM_LIB}" ]; then
             LIBSUFFIX="dylib"
             ;;
     esac
-    llvm-kompile $TEMPD/llvm-definition.kore $TEMPD/dt c -- \
-             -fPIC -std=c++17 -o $TEMPD/interpreter \
-             $PLUGIN_LIBS $PLUGIN_INCLUDE $PLUGIN_CPP \
-             -lcrypto -lssl $LPROCPS
-    lib=$TEMPD/interpreter.$LIBSUFFIX
+
+    if [ ! -f "$log_dir/$LLVM_DEFINITION_SHA.$LIBSUFFIX" ]; then
+        if [ ! -d "${PLUGIN_DIR}" ]; then
+            echo "Either LLVM_LIB or PLUGIN_DIR must be provided"
+            exit 2
+        fi
+        #generate matching data
+        (cd $TEMPD && mkdir -p dt && llvm-kompile-matching llvm-definition.kore qbaL ./dt 1/2)
+        # find library dependencies and source files
+        for lib in libff libcryptopp blake2; do
+            LIBFILE=$(find ${PLUGIN_DIR} -name "${lib}.a" | head -1)
+            [ -z "$LIBFILE" ] && (echo "[Error] Unable to locate ${lib}.a"; exit 1)
+            PLUGIN_LIBS+="$LIBFILE "
+            PLUGIN_INCLUDE+="-I$(dirname $LIBFILE)/../include "
+        done
+        PLUGIN_CPP="${PLUGIN_DIR}/include/plugin-c/blake2.cpp ${PLUGIN_DIR}/include/plugin-c/crypto.cpp ${PLUGIN_DIR}/include/plugin-c/plugin_util.cpp"
```

Comment:
Just noticed the left-over `blake2` source file here...
```suggestion
        PLUGIN_CPP="${PLUGIN_DIR}/include/plugin-c/crypto.cpp ${PLUGIN_DIR}/include/plugin-c/plugin_util.cpp"
```